lib/FreeRTOS/freertos-plus-mqtt/agent_command_pool.c

```diff
@@ -66,11 +66,19 @@ Command_t * Agent_GetCommand( uint32_t blockTimeMs )
 
     if( !initialized )
     {
-        memset( ( void * ) commandStructurePool, 0x00, sizeof( commandStructurePool ) );
-        freeCommandStructMutex = xSemaphoreCreateCounting( MQTT_COMMAND_CONTEXTS_POOL_SIZE, MQTT_COMMAND_CONTEXTS_POOL_SIZE );
-        configASSERT( freeCommandStructMutex ); /*_RB_ Create all objects here statically. */
+        taskENTER_CRITICAL();
```

Comment:
I think this check needs to be on line 66 since the static variable `initialized` is accessed on lie 67 as well.

Reply:
That is a good thought. I like that idea. However, `initialized` could be a volatile to avoid the race conditions that can occur due to compiler optimizations.